tool/tsh/db.go

```diff
@@ -290,6 +293,28 @@ Key:       %v
 	return nil
 }
 
+func startLocalALPNSNIProxy(cf *CLIConf, tc *client.TeleportClient, databaseProtocol string) (*alpnproxy.LocalProxy, error) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	lp, err := mkLocalProxy(cf.Context, tc.WebProxyAddr, databaseProtocol, listener)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	startSyncC := make(chan struct{})
+	go func() {
+		close(startSyncC)
```

Comment:
Right, the `startSyncC` was suppose to prevent  flaky test without time.Sleep call  where dial call is invoked immediately but retries calls seems to be move better solution.